include/dlaf/mc/cholesky_local.h

```diff
@@ -0,0 +1,117 @@
+// Linear Algebra with Future (DLAF)
+//
+// Copyright (c) 2018-2019, ETH Zurich
+// All rights reserved.
+//
+// Please, refer to the LICENSE file in the root directory.
+// SPDX-License-Identifier: BSD-3-Clause
+//
+#pragma once
+
+#include "dlaf/blas_tile.h"
+#include "dlaf/lapack_tile.h"
+#include "dlaf/matrix.h"
+#include "dlaf/matrix/index.h"
+#include "dlaf/matrix/layout_info.h"
+#include "dlaf/util_blas.h"
+#include "dlaf/util_matrix.h"
+//
+#include <exception>
+//
+#include <hpx/runtime/threads/run_as_hpx_thread.hpp>
+#include "hpx/hpx.hpp"
+#include "hpx/include/parallel_executors.hpp"
+#include "hpx/include/threads.hpp"
+
+/// @file
+
+namespace dlaf {
+
+/// @brief Cholesky implementation on local memory
+///
+/// @tparam mat referes to a dlaf::Matrix object
+/// @param uplo specifies whether the matrix is \a Upper or \a Lower triangular
+/// @param side defines whether the matrix is on the \a Left or \a Right side of the unknown matrix
+/// @param op specifies the form of \p mat used in matrix multiplication: \a NoTrans, \a Trans, \a ConjTrans
+/// @param diag describes if the matrix is unit triangular (\a Unit) or not (\a NonUnit)
+///
+/// @throws std::runtime_error if \p uplo = \a Upper decomposition  is chosen (not yet implemented)
+
+template <class T>
+void cholesky_local(Matrix<T, Device::CPU>& mat, blas::Uplo uplo, blas::Side side, blas::Op op,
+                    blas::Diag diag) {
```

Comment:
The signature should be:
`void cholesky_local(blas::Uplo uplo, Matrix<T, Device::CPU>& mat)`